clusterloader2/pkg/measurement/common/wait_for_controlled_pods.go

```diff
@@ -23,122 +23,397 @@ package common
 
 import (
 	"fmt"
+	"reflect"
+	"strconv"
+	"sync"
 	"sync/atomic"
 	"time"
 
 	"github.com/golang/glog"
 	batch "k8s.io/api/batch/v1"
-	"k8s.io/api/core/v1"
+	corev1 "k8s.io/api/core/v1"
 	extensions "k8s.io/api/extensions/v1beta1"
+	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/util/wait"
 	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
 	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
 	"k8s.io/perf-tests/clusterloader2/pkg/util"
 )
 
 const (
-	defaultWaitForRCPodsTimeout      = 60 * time.Second
-	defaultWaitForRCPodsInterval     = 5 * time.Second
-	waitForControlledPodsRunningName = "WaitForControlledPodsRunning"
+	defaultWaitForRCPodsTimeout        = 60 * time.Second
+	defaultOperationTimeout            = 10 * time.Minute
+	checkControlledPodsInterval        = 5 * time.Second
+	waitForControlledPodsRunningName   = "WaitForControlledPodsRunning"
+	waitForControlledPodsMaxGoroutines = 25
 )
 
 func init() {
 	measurement.Register(waitForControlledPodsRunningName, createWaitForControlledPodsRunningMeasurement)
 }
 
 func createWaitForControlledPodsRunningMeasurement() measurement.Measurement {
-	return &waitForControlledPodsRunningMeasurement{}
+	return &waitForControlledPodsRunningMeasurement{
+		checkerMap: make(map[string]*objectChecker),
+	}
 }
 
-type waitForControlledPodsRunningMeasurement struct{}
+type waitForControlledPodsRunningMeasurement struct {
+	informer          cache.SharedInformer
+	kind              string
+	namespace         string
+	labelSelector     string
+	fieldSelector     string
+	operationTimeout  time.Duration
+	stopCh            chan struct{}
+	isRunning         bool
+	runningRoutines   int32
+	lock              sync.Mutex
+	opResourceVersion uint64
+	checkerMap        map[string]*objectChecker
+}
```

Comment:
The problem with current implementation is that you can start only a single measurement.

If I would like to track different sets of objects at the same time, that's currently impossible.

Reply:
It is possible to have more then one instance of given measurement.